src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -147,21 +150,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
   /**
-   * Returns the Entity in database that has {@code url}, or
+   * Returns the Entity in database that has {@code videoId}, or
    * {@code Optional.empty()} if one doesn't exist.
    */
-  // TODO: Use a filter to avoid fetching all lectures.  See: #185.
-  private Optional<Entity> checkUrlInDatabase(String url) {
-    Query query = new Query(LectureUtil.KIND);
+  private Optional<Entity> checkVideoInDatabase(String videoId) {
```

Comment:
nit: I find it weird that the method name has the word "check" in it because at least to me, I feel that implies a boolean is returned. Would it be better to return a boolean instead of an optional considering `existingEntity` is only used for checking?

Reply:
Renamed to `queryForLectureWithVideoId`